types/vote.go

```diff
@@ -45,6 +46,52 @@ func NewConflictingVoteError(vote1, vote2 *Vote) *ErrVoteConflictingVotes {
 // Address is hex bytes.
 type Address = crypto.Address
 
+// VoteExtensionToSign is a subset of VoteExtension
+// that is signed by the validators private key
+type VoteExtensionToSign struct {
+	AppDataToSign []byte `json:"app_data_to_sign"`
+}
+
+// BytesPacked returns a bytes-packed representation for
+// debugging and human identification. This function should
+// not be used for any logical operations.
+func (ext VoteExtensionToSign) BytesPacked() []byte {
+	res := make([]byte, len(ext.AppDataToSign))
+	copy(res, ext.AppDataToSign)
+	return res
+}
+
+// ToVoteExtension constructs a VoteExtension from a VoteExtensionToSign
+func (ext VoteExtensionToSign) ToVoteExtension() VoteExtension {
+	return VoteExtension{
+		AppDataToSign: ext.AppDataToSign,
+	}
+}
+
+// VoteExtension is a set of data provided by the application
+// that is additionally included in the vote
+type VoteExtension struct {
+	AppDataToSign             []byte `json:"app_data_to_sign"`
+	AppDataSelfAuthenticating []byte `json:"app_data_self_authenticating"`
```

Comment:
I'm still not quite clear on what this field is or what it is to be used for? Is it a signature or something?

Reply:
Both AppData* are custom data returned by the app, but only the AppDataToSign is actually get signed by the validator. AppDataSelfAuthenticating could be used for aggregated signatures(or other signature extensions).

Reply:
For threshold decryption, `AppDataSelfAuthenticating` is going to be each validators decryption shares for txs in the block it is precommitting over. 

These should _not be_ signed over, as they will not get included in the final block data.

Every node can check this self-authenticating data against the validators public key shares in the state machine, and be convinced of the decryption share validity.